mico-core/src/main/java/io/github/ust/mico/core/persistence/MicoServiceRepository.java

```diff
@@ -47,28 +46,43 @@
     @Query("MATCH (application:MicoApplication)-[:INCLUDES_SERVICE]-(service:MicoService) WHERE application.shortName = {applicationShortName} AND application.version = {applicationVersion} RETURN COLLECT(service) AS services")
     List<MicoService> findAllByApplication(@Param("applicationShortName") String applicationShortName, @Param("applicationVersion") String applicationVersion);
 
-    /**
-     * Find a specific service interface.
-     *
-     * The returned interface will NOT have ports mapped by the ogm.
-     * If you want to have a interface with mapped ports use the serviceInterface
-     * list in the corresponding MicoService object returned by findByShortNameAndVersion!
-     *
-     * @param serviceInterfaceName
-     * @param shortName
-     * @param version
-     * @return
-     */
-    @Query("MATCH (service:MicoService)-[:PROVIDES_INTERFACES]->(interface:MicoServiceInterface)-[:PROVIDES_PORTS]->(port:MicoServicePort) WHERE service.shortName = {shortName} AND service.version = {version} AND interface.serviceInterfaceName = {serviceInterfaceName} return COLLECT(port) AS ports")
-    Optional<MicoServiceInterface> findInterfaceOfServiceByName(@Param("serviceInterfaceName") String serviceInterfaceName, @Param("shortName") String shortName, @Param("version") String version);
-
+    // Doesn't this belong in the MicoServiceInterfaceRepository?
```

Comment:
Is this a ToDo?

Reply:
Yes, but I think it's best to handle it in a different issue, I will create one.

Reply:
See #561.